sogs/events.py

```diff
@@ -0,0 +1,160 @@
+from collections import defaultdict
+
+from oxenmq import AuthLevel
+
+from . import model
+from .omq import omq
+from .web import app
+
+from binascii import hexlify
+from types import Iterable
+
+from oxenc import bt_serialize
+
+from .routes.subrequest import make_subrequest
+
+from flask import g
+
+# pools for event propagation
+_pools = defaultdict(set)
+
+status_OK = 'OK'
+status_ERR = 'ERROR'
+
+# the events we are able to subscribe to
+EVENTS = ('message', 'joined', 'parted', 'banned', 'unbanned', 'deleted', 'uploaded')
+
+
+def event_name_valid(eventname):
+    """ return True if this event name is something well formed """
+    return eventname in EVENTS
+
+
+def _user_from_conn(conn):
+    """
+    make a model.User from a connection using it's curve pubkey as the session id.
+    """
+    return model.User(session_id='05' + hexlify(conn.pubkey).decode())
```

Comment:
`bytes` have a `hex()` method:
```suggestion
    return model.User(session_id='05' + conn.pubkey.hex())
```